### PR TITLE
Linting

### PR DESCRIPTION
**Pull request by jrwells**
1) passes current linting rules
2) added a pre-commit linting hook

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -23,10 +23,13 @@
     "build:server": "NODE_ENV=production webpack --config ./webpack/server.babel.js",
     "build:db": "node ./src/server/database/setupDB.babel.js",
     "update:db": "node ./src/server/database/setupDB.babel.js true",
-    "test": "NODE_ENV=testing ava ./src/**/__tests___/**/*.js --verbose",
+    "test": "xo && NODE_ENV=testing ava ./src/**/__tests___/**/*.js --verbose",
```
Comment (jordanh) [inline package.json]: Did you mean to use `xo` here or did you really want to `npm lint` instead?

File: src/server/server.babel.js [hunk 2]
```diff
@@ -1,12 +1,14 @@
-if (process.env.NODE_ENV !== 'production') {
-  if (!require('piping')({
-    hook: false,
-    ignore: /(\/\.|~$|\.json$)/i
-  })) {
-    return;
+(() => {
```
Comment (jordanh) [inline src/server/server.babel.js]: I don't think this is quite Kosher. You're using an arrow function (ES6) in a file that needs to be straight up ecmascript 2015. I'd suggest you revert your changes and just inform the linter to ignore this file.

File: src/universal/modules/meeting/containers/MeetingLayout/MeetingLayout.js [hunk 3]
```diff
@@ -64,23 +64,29 @@ const mapStateToProps = state => {
...
     isAuthenticated: auth.get('isAuthenticated')
   };
-}
+};
 
-const socketClusterListeners = {
+const socketClusterListeners = { // eslint-disable-line no-unused-vars
   unsubscribe(props) {
-    debugger
+    debugger;
```
Comment (jordanh) [inline src/universal/modules/meeting/containers/MeetingLayout/MeetingLayout.js]: Thinking this line should be removed. Any reason this snuck in here @mattkrick?

File: src/universal/utils/schema.js [hunk 4]
```diff
@@ -4,8 +4,8 @@ export function parsedJoiErrors(error) {
   }
   const errors = {};
   const allErrors = error.details;
-  for (let i = 0; i < allErrors.length; i++) {
-    const curError = allErrors[i];
+  for (let index = 0; index < allErrors.length; index++) {
```
Comment (jordanh) [inline src/universal/utils/schema.js]: I prefer `idx` to `index` but all good ;)

File: src/universal/modules/meeting/containers/MeetingLayout/MeetingLayout.js (hunk 3 above)
Comment (mattkrick) [inline src/universal/modules/meeting/containers/MeetingLayout/MeetingLayout.js]: whoa, how'd that get in there? kill it with fire!

**Pull request closed without merging**